File: scripts/publish.js
import { $ } from 'bun'
import fs from 'fs/promises'
import path from 'path'
import { Glob } from 'bun'

// Tired of non functional workflows extensions
// Upload specifid file to latest release script

// Github options
const OWNER = 'thewh1teagle'
const REPO = 'vibe'
const TOKEN = process.env.GH_TOKEN
const tauriConfPath = path.join(__dirname, '../desktop/src-tauri/tauri.conf.json')
const TAG = 'v' + (await JSON.parse(await fs.readFile(tauriConfPath)).version)
const DST = process.argv[2]
const glob = new Glob(DST)
const DEBUG = process.env.DEBUG === '1'

async function publish(dst, token, tag) {
	await $`gh release upload ${tag} ${dst} --clobber`
}

for await (const file of glob.scan()) {
	await publish(file, TOKEN, TAG)
}

File: scripts/publish.py
#!/usr/bin/env -S uv run --script
# /// script
# requires-python = ">=3.12"
# ///

from __future__ import annotations

import glob
import json
import subprocess
import sys
from pathlib import Path

OWNER = "thewh1teagle"
REPO = "vibe"
TAURI_CONF_PATH = Path(__file__).resolve().parent.parent / "desktop/src-tauri/tauri.conf.json"


def publish(dst: str, tag: str) -> None:
    subprocess.run(["gh", "release", "upload", tag, dst, "--clobber"], check=True)


def main(argv: list[str]) -> int:
    if len(argv) < 2:
        print("Usage: uv run scripts/publish.py '<glob-pattern>'", file=sys.stderr)
        return 1

    config = json.loads(TAURI_CONF_PATH.read_text(encoding="utf-8"))
    tag = "v" + config["version"]
    dst = argv[1]

    for file_path in glob.glob(dst):
        publish(file_path, tag)
    return 0


if __name__ == "__main__":
    sys.exit(main(sys.argv))
